### PR TITLE
New Task topic: Update API Objects in Place Using kubectl patch

### PR DESCRIPTION
**Pull request by steveperry-53**
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes.github.io/4519)

### CODE REVIEW

File: docs/tasks/run-application/update-api-object-in-place.md [hunk 1]
```diff
@@ -0,0 +1,268 @@
...
+
+The output shows that the PodSpec in the Deployment has two Containers:
+
+```shell
+containers:
+- image: nginx
+  imagePullPolicy: Always
+  name: patch-demo-ctr
+  ...
+  - image: redis
```
Comment (tengqm) [inline docs/tasks/run-application/update-api-object-in-place.md]: will line 88 cause indentation problem?

File: docs/tasks/run-application/update-api-object-in-place.md [hunk 2]
```diff
@@ -0,0 +1,268 @@
...
+
+## Use a JSON merge patch to update a Deployment
+
+A strategic merge patch is different from a
+[JSON merge patch](https://tools.ietf.org/html/rfc6902).
+With a JSON merge patch, if you
+want to update a list, you have to specify the entire new list. And the new list completely
+replaces the existing list.
+
+The `kubectl patch` command has a `type` parameter that you can set to to one of these values:
```
Comment (tengqm) [inline docs/tasks/run-application/update-api-object-in-place.md]: "set to to" -> "set to"

File: docs/tasks/run-application/update-api-object-in-place.md [hunk 3]
```diff
@@ -0,0 +1,268 @@
...
+  <tr><td>json</td><td><a href="https://tools.ietf.org/html/rfc6902">JSON Patch, RFC 6902</a></td></tr>
+  <tr><td>merge</td><td><a href="https://tools.ietf.org/html/rfc7386">JSON Merge Patch, RFC 7386</a></td></tr>
+  <tr><td>strategic</td><td>Strategic merge patch</td></tr>
+</table>
+
+The default value for the `type` parameter is `strategic`. So in the preceding exercise, you
+did a strategic merge patch.
+
+Next, do a JSON merge patch on your same Deployment. Create a file named patch-file-2.yaml
+that has this content:
```
Comment (tengqm) [inline docs/tasks/run-application/update-api-object-in-place.md]: patch-file-2.yaml -> `patch-file-2.yaml`

File: docs/tasks/run-application/update-api-object-in-place.md [hunk 4]
```diff
@@ -0,0 +1,268 @@
...
+[JSON merge patch](https://tools.ietf.org/html/rfc6902).
+With a JSON merge patch, if you
+want to update a list, you have to specify the entire new list. And the new list completely
+replaces the existing list.
+
+The `kubectl patch` command has a `type` parameter that you can set to to one of these values:
+
+<table>
+  <tr><th>Parameter value</th><th>Merge type</th></tr>
+  <tr><td>json</td><td><a href="https://tools.ietf.org/html/rfc6902">JSON Patch, RFC 6902</a></td></tr>
```
Comment (tengqm) [inline docs/tasks/run-application/update-api-object-in-place.md]: Better add one sentence or two explaining how 'JSON' patch differs from 'JSON merge patch'.

File: docs/concepts/cluster-administration/manage-deployment.md [hunk 5]
```diff
@@ -308,27 +308,6 @@ For more information, please see [kubectl scale](/docs/user-guide/kubectl/{{page
 
 Sometimes it's necessary to make narrow, non-disruptive updates to resources you've created.
 
-### kubectl apply
```
Comment (steveperry-53) [inline docs/concepts/cluster-administration/manage-deployment.md]: Fixed.

File: docs/tasks/run-application/update-api-object-in-place.md [hunk 6]
```diff
@@ -0,0 +1,268 @@
+---
+title: Update API Objects in Place Using kubectl patch
```
Comment (steveperry-53) [inline docs/tasks/run-application/update-api-object-in-place.md]: I want to keep this under Tasks, not Concepts. Yes, I think we should create other Update tasks and consolidate them under a heading. For now, let's leave it where it is.

I changed the filename to update-api-object-kubectl-patch.md

File: docs/tasks/run-application/update-api-object-in-place.md (hunk 1 above)
Comment (steveperry-53) [inline docs/tasks/run-application/update-api-object-in-place.md]: Fixed.

File: docs/tasks/run-application/update-api-object-in-place.md (hunk 2 above)
Comment (steveperry-53) [inline docs/tasks/run-application/update-api-object-in-place.md]: Fixed

File: docs/tasks/run-application/update-api-object-in-place.md (hunk 3 above)
Comment (steveperry-53) [inline docs/tasks/run-application/update-api-object-in-place.md]: Fixed

File: docs/tasks/run-application/update-api-object-in-place.md (hunk 4 above)
Comment (steveperry-53) [inline docs/tasks/run-application/update-api-object-in-place.md]: Added a link to a comparison article.

File: docs/tasks/run-application/update-api-object-in-place.md (hunk 6 above)
Comment (steveperry-53) [inline docs/tasks/run-application/update-api-object-in-place.md]: Currently, [Managing Resources](https://kubernetes.io/docs/concepts/cluster-administration/manage-deployment/) is a mix of conceptual material and several tasks. I'l like to separate out the tasks into their own topics. Maybe under Tasks, we'll want to have a bucket for tasks about updating.

**Comment by steveperry-53** [comment]
@tengqm, Thanks for the review. I've addressed your comments in https://github.com/kubernetes/kubernetes.github.io/pull/4519/commits/3a4825e51c7d02b58fc7fb763f3ef25d649f5f51.

**Pull request merged**